pkg/alertmanager/alertmanager.go

```diff
@@ -241,7 +242,38 @@ func New(cfg *Config, reg *prometheus.Registry) (*Alertmanager, error) {
 		am.wg.Done()
 	}()
 
-	am.alerts, err = mem.NewAlerts(context.Background(), am.marker, 30*time.Minute, nil, am.logger)
+	var callback mem.AlertStoreCallback
+	if am.cfg.Limits != nil {
+		insertAlertFailures := promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Name: "alertmanager_alerts_insert_failures_total",
+			Help: "Number of failures to insert new alerts to in-memory alert store.",
+		}, []string{"reason"})
```

Comment:
This has reason but when we expose it via `cortex_alertmanager_alerts_insert_failures_total ` we loose it.

Reply:
I've removed the reason from "internal" metric. While it was nice to have, without exposing it to "aggregated" registry, it is useless, and we already expose current alert count and alerts size metrics to help troubleshooting.